### PR TITLE
chore(ci): use github actions

### PR DESCRIPTION
**Pull request by RobinKnipe**
Handle basic code quality assurance checking (CI) with GitHub Actions.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean install of node dependencies, build the source code
+# and run tests across different versions of node
+# For more information see:
+# https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ master ]
```
Comment (daniel-ac-martin) [inline .github/workflows/node.js.yml:10]: Are you sure you want this clause?

Comment (daniel-ac-martin) [reply .github/workflows/node.js.yml:10]: i.e. Why not pushes to all branches?

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x]
```
Comment (daniel-ac-martin) [inline .github/workflows/node.js.yml:21]: We should probably move to v12 and forget about v10.

However we would need to migrate to a v12 base-image for that.

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    - run: npm run build --if-present
```
Comment (daniel-ac-martin) [inline .github/workflows/node.js.yml:30]: I'd remove this. - We don't have a `build` script and if we did it wouldn't necessarily be a prerequisite of `npm test`.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (RobinKnipe) [reply .github/workflows/node.js.yml]: Agreed, fixed

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (RobinKnipe) [reply .github/workflows/node.js.yml]: Switched to use `node12` only. The recent changes (rebased) introduce a node 12 container, so we should be good...

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (RobinKnipe) [reply .github/workflows/node.js.yml]: Agreed, removed

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (daniel-ac-martin) [reply .github/workflows/node.js.yml]: It's nice to at least unit test on 14 as well. They are quick to run and they might flag up future problems.

Comment (daniel-ac-martin) [reply .github/workflows/node.js.yml]: But that can be sorted in another PR.

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# This workflow will do a clean install of node dependencies, build the source code
+# and run tests across different versions of node
+# For more information see:
+# https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on: [push, pull_request  ]
```
Comment (daniel-ac-martin) [inline .github/workflows/node.js.yml:8]: Whitespace is a bit funky here.

**Review by daniel-ac-martin — DISMISSED** [review]
Good enough to merge but check my comments and, if you agree, either fix or add them as issues to be fixed later.

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (RobinKnipe) [reply .github/workflows/node.js.yml]: Fixed

File: .github/workflows/node.js.yml (hunk 4 above)
Comment (RobinKnipe) [reply .github/workflows/node.js.yml]: Fixed

**Review by daniel-ac-martin — APPROVED** [review]

**Pull request merged**